techlibs/xilinx/synth_xilinx.cc

```diff
@@ -269,16 +281,36 @@ struct SynthXilinxPass : public Pass
 
 		if (check_label(active, run_from, run_to, "map_cells"))
 		{
+			if (!nosrl) {
+				// shregmap operates on bit-level flops, not word-level,
+				//   so break those down here
+				Pass::call(design, "simplemap t:$dff t:$dffe");
+				// shregmap -tech xilinx can cope with $shiftx and $mux
+				//   cells for identifiying variable-length shift registers,
+				//   so attempt to convert $pmux-es to the former
+				Pass::call(design, "pmux2shiftx");
```

Comment:
That's very late for running `pmux2shiftx`. Look at all the fine-grain stuff that happened already at this point. Are there even any $eq cells left at this stage? This should be run at about the same time as `wreduce`, or `alumacc`, during the coarse-grain phase. Any word-level structures that are still preserved at this point are just preserved accidentally, not because there'd be any guarantee.

Reply:
I've made a number of (big) changes to the flow such that coarse grained/word-level should be preserved right up to `abc` being called. I'm hesitant to move it up to the coarse-grained level since (currently) `shregmap` can only analyse bit-level DFFs, but I'm open to discussing the pros and cons of each.

Reply:
Fair enough. Approving it. You will be the one who has to deal with the fallout if this isn't working as expected anyway.   ;)